libs/db/badgerdb/db.go

```diff
@@ -0,0 +1,288 @@
+package badgerdb
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/dgraph-io/badger/v3"
+
+	tmdb "github.com/lazyledger/lazyledger-core/libs/db"
```

Comment:
Not sure the alias makes much sense. Also isn't this sometimes called `dbm` / `tmdb`? Maybe something that signals that these are interfaces?

Reply:
`dbi` sounds good to me. But we can also brush this in a separate PR IMO. 

I think the only thing left to clarify is: https://github.com/lazyledger/lazyledger-core/pull/241#discussion_r598602931

Everything else looks good. I'll give it another pass shortly.